### ISSUE TITLE
git push fails; lfs sending to wrong remote repo

### ISSUE THREAD

**Issue by rcdailey** [git-lfs/git-lfs#3561]
I ran `git lfs install` in one of my submodules. I added a bunch of binary files. I then do:

```
git push -f origin
```

Relevant git configuration:

```
[push]
    default = current
[pull]
    rebase = true
[remote]
    pushDefault = fork
```

I have the following remotes (note: `company` is the *redacted* name of the repository organization):

```
$ git remote -v
fork    git@bitbucket.org:rcdailey/core.git (fetch)
fork    git@bitbucket.org:rcdailey/core.git (push)
origin  git@bitbucket.org:company/core.git (fetch)
origin  git@bitbucket.org:company/core.git (push)
```

Note: I'm using SSH protocol for my repository remote URLs.

Result of `git lfs env` is:

```
$ git lfs env
git-lfs/2.7.1 (GitHub; windows amd64; go 1.11.5; git 6b7fb6e3)
git version 2.20.1.windows.1

Endpoint=https://bitbucket.org/company/core.git/info/lfs (auth=none)
  SSH=git@bitbucket.org:company/core.git
Endpoint (pushdefault)=https://bitbucket.org/company/core.git/info/lfs (auth=none)
  SSH=git@bitbucket.org:company/core.git
Endpoint (fork)=https://bitbucket.org/rcdailey/core.git/info/lfs (auth=none)
  SSH=git@bitbucket.org:rcdailey/core.git
LocalWorkingDir=E:\code\frontend\source\Core
LocalGitDir=E:\code\frontend\source\.git\modules\Core
LocalGitStorageDir=E:\code\frontend\source\.git\modules\Core
LocalMediaDir=E:\code\frontend\source\.git\modules\Core\lfs\objects
LocalReferenceDirs=
TempDir=E:\code\frontend\source\.git\modules\Core\lfs\tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=E:\code\frontend\source\.git\modules\Core\lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic
UploadTransfers=basic
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
GIT_LFS_PATH=C:\Program Files\Git LFS
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

And the output of my push command, with `GIT_TRACE` enabled:

```
$ GIT_TRACE=1 git push -f origin
10:41:12.121661 exec-cmd.c:236          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
10:41:12.125662 git.c:418               trace: built-in: git push -f origin
10:41:12.134657 run-command.c:643       trace: run_command: unset GIT_DIR GIT_PREFIX; ssh git@bitbucket.org 'git-receive-pack '\''company/core.git'\'''
10:41:13.511567 run-command.c:643       trace: run_command: E:/code/frontend/source/.git/modules/Core/hooks/pre-push origin git@bitbucket.org:company/core.git
10:41:13.585556 exec-cmd.c:236          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
10:41:13.588542 git.c:675               trace: exec: git-lfs pre-push origin git@bitbucket.org:company/core.git
10:41:13.588542 run-command.c:643       trace: run_command: git-lfs pre-push origin git@bitbucket.org:company/core.git
10:41:13.624657 trace git-lfs: exec: git 'version'
10:41:13.667704 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
10:41:13.733841 trace git-lfs: exec: git 'config' '-l'
10:41:13.755816 trace git-lfs: pre-push: refs/heads/feature/lfs 53e51ee3aa4907613875799610b32cf0c6cfe3da refs/heads/feature/lfs 575024471d8a95cd38a79b4b8e3b7229f8f97dc8
10:41:14.789926 trace git-lfs: tq: running as batched queue, batch size of 100
10:41:14.790889 trace git-lfs: run_command: git rev-list --stdin --objects --
10:41:15.659979 trace git-lfs: tq: sending batch of size 100
10:41:15.659979 trace git-lfs: run_command: ssh -- git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload
10:41:16.253958 trace git-lfs: api: batch 100 files
10:41:16.263937 trace git-lfs: HTTP: POST https://bitbucket.org/%7B%7D/%7B60959e6c-fa17-4af4-be62-7066b1ea0b43%7D/info/lfs/objects/batch
10:41:20.155462 trace git-lfs: HTTP: 402
10:41:20.155462 trace git-lfs: HTTP: {"message": "\n********************************************************************************\n[ERROR] Your LFS push failed because you're out of file storage\n[ERROR] Change your plan to get more file storage:\n[ERROR] https://bitbucket.org/account/user/rcdailey/plans/\n********************************************************************************\n    "}
10:41:20.156411 trace git-lfs: api error:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

10:41:20.156411 trace git-lfs: tq: sending batch of size 100
10:41:20.156411 trace git-lfs: ssh cache: git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload
10:41:20.157409 trace git-lfs: api: batch 100 files
10:41:20.158408 trace git-lfs: HTTP: POST https://bitbucket.org/%7B%7D/%7B60959e6c-fa17-4af4-be62-7066b1ea0b43%7D/info/lfs/objects/batch
10:41:23.617935 trace git-lfs: HTTP: 402
10:41:23.617935 trace git-lfs: HTTP: {"message": "\n********************************************************************************\n[ERROR] Your LFS push failed because you're out of file storage\n[ERROR] Change your plan to get more file storage:\n[ERROR] https://bitbucket.org/account/user/rcdailey/plans/\n********************************************************************************\n    "}
10:41:23.617935 trace git-lfs: api error:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

10:41:23.617935 trace git-lfs: tq: sending batch of size 36
10:41:23.617935 trace git-lfs: ssh cache: git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload
10:41:23.618923 trace git-lfs: api: batch 36 files
10:41:23.618923 trace git-lfs: HTTP: POST https://bitbucket.org/%7B%7D/%7B60959e6c-fa17-4af4-be62-7066b1ea0b43%7D/info/lfs/objects/batch
10:41:27.140402 trace git-lfs: HTTP: 402
10:41:27.140402 trace git-lfs: HTTP: {"message": "\n********************************************************************************\n[ERROR] Your LFS push failed because you're out of file storage\n[ERROR] Change your plan to get more file storage:\n[ERROR] https://bitbucket.org/account/user/rcdailey/plans/\n********************************************************************************\n    "}
10:41:27.140402 trace git-lfs: api error:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

Uploading LFS objects:   0% (0/101), 0 B | 0 B/s, done
batch response:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

batch response:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

batch response:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

error: failed to push some refs to 'git@bitbucket.org:company/core.git'
```

One thing that catches my eye about the verbose logs during the push is:

    run_command: ssh -- git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload

Shouldn't it be `company/core.git`??

Basically, it is pushing to the *wrong remote URL*. It should be pushing to `origin`, not `fork`. Why is this broken? I did find [this issue](https://github.com/git-lfs/git-lfs/issues/2915), which seems related, since the `git lfs env` command does not show the `origin` remote (but it does show `fork`). However that is labeled to have been fixed in v2.5 of Git LFS and I'm using 2.7. I'm stuck, help!

**Comment by bk2204** [comment]
Hey, thanks for writing in.

I think this is indeed a bug. It looks like we're setting the default remote, but not the default push remote, which leads us to use the values specified by `remote.pushDefault`, which is `fork`.

I have a branch, `push-remote-set`, that will likely fix this issue. Could you try it out and see if it fixes your issue? You'll need to clone the main repository, add my clone as an additional remote, pull down the branch, and run `make`. You don't want to use `go get` or `go install` directly on my clone.

**Comment by rcdailey** [comment]
Unfortunately I'm using Git for Windows (on Windows platform). I do not believe I'll be able to easily build your project with your given instructions. If you use CMake that would be more feasible, but I haven't seen if you do.

Any way you could provide a prebuilt binary? I realize that puts a burden on you; I apologize for that.

**Comment by bk2204** [comment]
Sure. I'm attaching a 32-bit and 64-bit binary (in zip files, since I can't upload raw binaries). Be aware that these binaries are not signed, unlike the ones distributed with Git for Windows.

You can ignore the weird version number; they are in fact up to date with the latest `master` branch.

[git-lfs-windows-386-v2.7.0-23-g003ff89f.zip](https://github.com/git-lfs/git-lfs/files/2967520/git-lfs-windows-386-v2.7.0-23-g003ff89f.zip)
[git-lfs-windows-amd64-v2.7.0-23-g003ff89f.zip](https://github.com/git-lfs/git-lfs/files/2967521/git-lfs-windows-amd64-v2.7.0-23-g003ff89f.zip)

**Comment by rcdailey** [comment]
Thanks a bunch, and once again apologize for making extra work for you.

Looks like files are uploading now! I do not see any upload progress  when I run the command though, not sure why that is, or if it's a related issue.

**Comment by rcdailey** [comment]
Looks like after all files are uploaded, you see this printed instantly after:

```
Uploading LFS objects: 100% (236/236), 1.1 GB | 9.3 MB/s, done
```

But it doesn't show progress during the operation.

**Comment by bk2204** [comment]
Great, glad it's working now. I'll try to get a PR opened with that fix.

I'm not sure why you're seeing the delay in printing; usually we should start printing progress if the operation takes longer than a certain amount of time. It's possible that there's some weirdness because I built this on a Mac instead of a Windows box as we normally do for Windows versions.

**Comment by rcdailey** [comment]
Perfect, and thanks again.

I'm not sure when your next release will be, but I will make sure to test it again and open another bug if I'm still seeing the status update issue. Until then, I'll go ahead and close this issue.

**Comment by rcdailey** [comment]
@bk2204 I installed v2.7.2 and this isn't working. Shouldn't it have made it into that release?

**Comment by bk2204** [comment]
No, usually our point releases fix only a small number of bugs, most of which are regressions. This should, however, make it into 2.8.0, which should be released sometime in June.
